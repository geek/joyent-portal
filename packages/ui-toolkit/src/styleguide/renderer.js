import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
