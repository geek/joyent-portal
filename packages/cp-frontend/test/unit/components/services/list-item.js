
/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ServiceListItem from '@components/services/list-item.js';
import { Router, service } from '../../mocks';

it('renders <ServiceListItem /> without throwing', () => {
  const tree = renderer
    .create(
      <Router>
        <ServiceListItem service={service} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders child <ServiceListItem /> without throwing', () => {
  const tree = renderer
    .create(
      <ServiceListItem service={service} isChild />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
