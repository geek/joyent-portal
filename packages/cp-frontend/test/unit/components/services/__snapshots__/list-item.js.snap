// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <ServiceListItem /> without throwing 1`] = `
.c7 {
  color: ;
  color: ;
  text-decoration: none;
}

.c14 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  min-width: 7.5rem;
  box-sizing: border-box;
  display: inline-block;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0.9375rem 1.125rem;
  position: relative;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  font-size: 0.9375rem;
  text-align: center;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-decoration: none;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  color: ;
  background-image: none;
  background-color: ;
  border-radius: 0.25rem;
  border: solid 0.0625rem;
  border-radius: 0;
}

.c14::-moz-focus-inner,
.c14[type="button"]::-moz-focus-inner,
.c14[type="reset"]::-moz-focus-inner,
.c14[type="submit"]::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c14:-moz-focusring,
.c14[type="button"]:-moz-focusring,
.c14[type="reset"]:-moz-focusring,
.c14[type="submit"]:-moz-focusring {
  outline: 0.0625rem dotted ButtonText;
}

.c14:focus {
  outline: 0;
  text-decoration: none;
  background-color: ;
  border-color: ;
}

.c14:hover {
  background-color: ;
  border: solid 0.0625rem;
}

.c14:active,
.c14:active:hover,
.c14:active:focus {
  background-image: none;
  outline: 0;
  background-color: ;
  border-color: ;
}

.c14[disabled] {
  cursor: not-allowed;
  pointer-events: none;
}

.c14 + button {
  margin-left: 1.25rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.5;
  color: ;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-grow: 2;
  -ms-flex-grow: 2;
  flex-grow: 2;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
  width: 100%;
  padding: 0.75rem 1.125rem 0 1.125rem;
  color: ;
  -webkit-flex-grow: 6;
  -ms-flex-grow: 6;
  flex-grow: 6;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1.125rem;
}

.c19 {
  font-size: 0.9375rem;
  line-height: 1.5;
  color: ;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-grow: 2;
  -ms-flex-grow: 2;
  flex-grow: 2;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
  width: 100%;
  padding: 0.75rem 1.125rem 0 1.125rem;
}

.c8 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
}

.c18 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
  padding-bottom: 0.75rem;
}

.c9 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
  position: relative;
  height: auto;
  min-height: 7.875rem;
  margin-bottom: 0.625rem;
  border: 0.0625rem solid;
  background-color: ;
  box-shadow: 0 0.125rem 0 0 rgba(0,0,0,0.05);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 0rem;
  margin-left: 0rem;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
  position: relative;
  height: auto;
  min-height: 7.875rem;
  margin-bottom: 0.625rem;
  border: 0.0625rem solid;
  background-color: ;
  box-shadow: 0 0.125rem 0 0 rgba(0,0,0,0.05);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 0rem;
  margin-left: 0rem;
  min-height: auto;
  height: 3rem;
  margin-bottom: 1rem;
  box-shadow: 0 0.125rem 0 0 rgba(0,0,0,0.1);
}

.c2 {
  position: absolute;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: ;
  border: solid 0.0625rem;
  box-shadow: none;
  width: calc(100% + 0.125rem);
  margin: -0.0625rem -0.0625rem 0 -0.0625rem;
}

.c17 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0;
  height: auto;
  padding-top: 0;
  min-width: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c12 {
  display: block;
  -webkit-flex: 0 0 2.9375rem;
  -ms-flex: 0 0 2.9375rem;
  flex: 0 0 2.9375rem;
  border-left: 0.0625rem solid;
  box-sizing: border-box;
  border-left-color: ;
}

.c13 {
  position: relative;
  border-width: 0;
  box-shadow: none;
  width: 100%;
  min-width: 2.875rem !important;
  height: 7.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: visible;
  overflow-y: visible;
  height: 2.875rem;
}

.c13:focus {
  border-width: 0;
}

.c13:hover {
  border-width: 0;
}

.c13:active,
.c13:active:hover,
.c13:active:focus {
  border-width: 0;
}

.c15 {
  position: absolute;
  left: 50%;
  top: 35%;
}

.c16 {
  margin: 0 0 0.125rem -0.125rem;
  border-radius: 50%;
  background-color: ;
  width: 0.25rem;
  height: 0.25rem;
}

.c11 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  font-size: 0.9375rem;
  font-style: normal;
  font-stretch: normal;
  color: ;
  display: inline-block;
  color: undefined;
  margin-left: 1.5rem;
}

.c11::first-letter {
  text-transform: capitalize;
}

.c10 {
  position: absolute;
}

.c10 > svg {
  fill: undefined;
}

.c20 {
  display: inline-block;
  margin: 0 0 0.9375rem 0;
  height: 3.375rem;
  width: 12.5rem;
}

.c21 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  line-height: 1.5rem;
  font-size: 0.9375rem;
  margin: 0 0 0.375rem 0;
  font-size: 0.8125rem;
  line-height: 0.8125rem;
}

.c1 {
  position: relative;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  color: ;
}

<div
  className="c0"
  name="card"
>
  <div
    className="c1 c2 c3"
    name="card-header"
  >
    <div
      className="c4"
      name="container"
    >
      <div
        className="c5"
      >
        <a
          active={undefined}
          className="c6 c7"
          href="/deployment-groups//services/nginx"
          onClick={[Function]}
          secondary={true}
        >
          Nginx
        </a>
      </div>
    </div>
    <div
      className="c8 c4"
      name="container"
    >
      <div
        className="c9"
      >
        <div>
          <div
            className="c10"
            color="disabled"
          >
            <svg
              className=""
              height="9"
              viewBox="0 0 18 9"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                icon: instances
              </title>
              <path
                d="M4.5 0a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9m8.59 0c-.375 0-.739.047-1.09.12 2.184.456 3.818 2.239 3.818 4.38S14.184 8.424 12 8.88c.351.073.715.12 1.09.12C15.803 9 18 6.985 18 4.5S15.802 0 13.09 0M14 4.5C14 6.985 11.802 9 9.09 9c-.375 0-.739-.047-1.09-.12 2.184-.456 3.818-2.239 3.818-4.38S10.184.576 8 .12C8.351.047 8.715 0 9.09 0 11.803 0 14 2.015 14 4.5"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <label
            className="c11"
            color="disabled"
          >
            1 instance
          </label>
        </div>
      </div>
    </div>
    <nav
      className="c12"
      name="card-options"
    >
      <button
        className="c13 c14"
        onClick={[Function]}
      >
        <div
          className="c15"
        >
          <div
            className="c16"
          />
          <div
            className="c16"
          />
          <div
            className="c16"
          />
        </div>
      </button>
    </nav>
  </div>
  <div
    className="c17"
    name="card-view"
  >
    <div
      className="c18 c19"
      name="container"
    >
      <div
        className="c9"
      >
        <div
          className="c20"
        >
          <p
            className="c21"
          >
            1
                        instance
                        running
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders child <ServiceListItem /> without throwing 1`] = `
.c2 {
  font-size: 0.9375rem;
  line-height: 1.5;
  color: ;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-grow: 2;
  -ms-flex-grow: 2;
  flex-grow: 2;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
  width: 100%;
  padding: 0.75rem 1.125rem 0 1.125rem;
}

.c3 {
  display: inline-block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-basis: 5.625rem;
  -ms-flex-basis: 5.625rem;
  flex-basis: 5.625rem;
  padding-bottom: 0.75rem;
}

.c7 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
  position: relative;
  height: auto;
  min-height: 7.875rem;
  margin-bottom: 0.625rem;
  border: 0.0625rem solid;
  background-color: ;
  box-shadow: 0 0.125rem 0 0 rgba(0,0,0,0.05);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 0rem;
  margin-left: 0rem;
  box-shadow: none;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0;
  height: auto;
  padding-top: 0;
  min-width: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  display: inline-block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  font-size: 0.875rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c4 {
  display: inline-block;
  padding: 0 1.125rem;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
}

.c8 {
  display: inline-block;
  margin: 0 0 0.9375rem 0;
  height: 3.375rem;
  width: 12.5rem;
}

.c9 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  line-height: 1.5rem;
  font-size: 0.9375rem;
  margin: 0 0 0.375rem 0;
  font-size: 0.8125rem;
  line-height: 0.8125rem;
}

<div
  className="c0"
  name="card"
>
  <div
    className="c1"
    name="card-view"
  >
    <div
      className="c2"
      name="container"
    >
      <span
        className="c3"
      >
        Nginx
      </span>
    </div>
    <div
      className="c4 c2"
      name="container"
    >
      <span
        className="c5"
      >
        1
         
        instance
      </span>
    </div>
    <div
      className="c6 c2"
      name="container"
    >
      <div
        className="c7"
      >
        <div
          className="c8"
        >
          <p
            className="c9"
          >
            1
                        instance
                        running
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
